### ISSUE TITLE
Can't run record_codes.py 

### ISSUE THREAD

**Issue by lunxer** [crleblanc/hackPump#1]
Ran ./record_codes.py -d /dev/ttyACM0 -o ir_codes.json
and was able to record values fine, but not able to re-run this.

does nothing until crtl+c 

(and displays)

^CTraceback (most recent call last):
  File "./record_codes.py", line 73, in <module>
    main()
  File "./record_codes.py", line 64, in main
    codes = recordCodes(args.device)
  File "./record_codes.py", line 39, in recordCodes
    toy = IrToy(serialDevice)
  File "/usr/local/lib/python2.7/dist-packages/irtoy.py", line 39, in **init**
    self._setSamplingMode()
  File "/usr/local/lib/python2.7/dist-packages/irtoy.py", line 54, in _setSamplingMode
    self.protocolVersion = self.toy.read(3)
  File "/usr/local/lib/python2.7/dist-packages/serial/serialposix.py", line 446, in read
    ready,_,_ = select.select([self.fd],[],[], self._timeout)
KeyboardInterrupt

^C

Removed hackpump dir, re-ran git clone, but still never able to rerun the record command.

**Comment by crleblanc** [comment]
Hi Luxer,

I think the fact that this was working and suddenly stopped indicates a hardware problem.  The script doesn't store any state between runs, so it's almost certainly not in the python script.

Does the problem go away if you re-flash the IR Toy?  How about rebooting the machine?

PS: I doubt this would work unless you were running firmware revision 22, but is that the firmware version you're running?

Thanks,
Chris

**Comment by lunxer** [comment]
Hello! 
Thank you for replying.
I tried to disconnect the ir toy, and reconnect it and then i was able to run the script again, and with great success: Hurrah!

"Unfortunately" the cold weather have ended for this year so i keep the heat pump off until the fall, i appreciate the good instructions to get it running.
I am a avid google cal user so this is an great setup for me, however i going to try and get a 1 wire temp sensor in my living room and let the pump run at higher temps until the living room target temp is met.

tl:dr : disconnected the ir toy, reconnect, script works again

**Comment by crleblanc** [comment]
Seems like the underlying cause is in the IR toy firmware, as discussed in this issue: https://github.com/crleblanc/PyIrToy/issues/2. Will close issue.
